### PR TITLE
Allow 404 on DeleteComment

### LINKED ISSUE #13306: Crier races when trying to report prowjobs on a same PR concurrently
/assign @krzyzacy 
/area prow
/area prow/crier

```yaml
{
 insertId:  "nbkepcg1aiz7s2"  
 jsonPayload: {
  component:  "crier"   
  error:  "error deleting comment: status code 404 not one of [204], body: {"message":"Not Found","documentation_url":"https://developer.github.com/v3/issues/comments/#delete-a-comment"}"   
  file:  "prow/crier/controller.go:254"   
  func:  "k8s.io/test-infra/prow/crier.(*Controller).processNextItem"   
  jobName:  "bff60437-9cfa-11e9-bdac-7e3a520fb0ef"   
  jobStatus: {…}   
  level:  "error"   
  msg:  "failed to report job"   
  prowjob:  "default/bff60437-9cfa-11e9-bdac-7e3a520fb0ef"   
 }
 labels: {…}  
 logName:  "projects/k8s-prow/logs/crier"  
 receiveTimestamp:  "2019-07-02T18:55:44.312460164Z"  
 resource: {…}  
 severity:  "ERROR"  
 timestamp:  "2019-07-02T18:55:40Z"  
}
```

```yaml
{
 insertId:  "10cys94g1ao1enz"  
 jsonPayload: {
  component:  "crier"   
  error:  "error deleting comment: status code 404 not one of [204], body: {"message":"Not Found","documentation_url":"https://developer.github.com/v3/issues/comments/#delete-a-comment"}"   
  file:  "prow/crier/controller.go:254"   
  func:  "k8s.io/test-infra/prow/crier.(*Controller).processNextItem"   
  jobName:  "6535ce25-9cfa-11e9-bdac-7e3a520fb0ef"   
  jobStatus: {…}   
  level:  "error"   
  msg:  "failed to report job"   
  prowjob:  "default/6535ce25-9cfa-11e9-bdac-7e3a520fb0ef"   
 }
 labels: {…}  
 logName:  "projects/k8s-prow/logs/crier"  
 receiveTimestamp:  "2019-07-02T18:53:20.148533634Z"  
 resource: {…}  
 severity:  "ERROR"  
 timestamp:  "2019-07-02T18:53:13Z"  
}
```

### PR DESCRIPTION
**Pull request by krzyzacy**
fixes #13306

/assign @fejta @cjwagner @stevekuznetsov

### CODE REVIEW

**Comment by krzyzacy** [comment]
actually it is crier trying to report the same prowjob on multiple threads at the same time.

@stevekuznetsov @munnerz client-go question here: I thought the cache key is unique? I guess while we processing a key, and we got an update event for the same key, it will process that key again?

File: prow/github/client.go [hunk 1]
```diff
@@ -1086,11 +1086,15 @@ func (c *client) CreateComment(org, repo string, number int, comment string) err
...
 func (c *client) DeleteComment(org, repo string, id int) error {
 	c.log("DeleteComment", org, repo, id)
-	_, err := c.request(&request{
+	rc, err := c.request(&request{
 		method:    http.MethodDelete,
 		path:      fmt.Sprintf("/repos/%s/%s/issues/comments/%d", org, repo, id),
 		exitCodes: []int{204},
 	}, nil)
+	if rc == 404 {
+		c.log("DeleteComment: getting 404, the comment might already got deleted", org, repo, id)
```
Comment (stevekuznetsov) [inline prow/github/client.go]: Is this at info level?

Comment (krzyzacy) [reply prow/github/client.go]: https://github.com/kubernetes/test-infra/blob/fbf31f75c8b562c8e12f8e340d2d1e64e0b0f857/prow/github/client.go#L489

**Comment by alvaroaleman** [comment]
> @stevekuznetsov @munnerz client-go question here: I thought the cache key is unique? I guess while we processing a key, and we got an update event for the same key, it will process that key again?

It is unique and yes, we process it again instantly after the first processing finishes. Maybe https://github.com/kubernetes/test-infra/pull/13351 is the fix for the underlying issue in crier?

**Comment by krzyzacy** [comment]
@alvaroaleman I think the real issue is the one on slack (report multiple prowjobs on the same PR at the same time)

I'd keep this hold for a bit until we have a good solution.

File: prow/github/client.go (hunk 1 above)
Comment (fejta) [reply prow/github/client.go]: might have already gotten (although maybe got might be correct in the UK)

Comment (munnerz) [reply prow/github/client.go]: 😅 FWIW (as I say 'gotten' all the time), my mum tells me that gotten isn't a real word in the UK 😱 😆

**Comment by munnerz** [comment]
> It is unique and yes, we process it again instantly after the first processing finishes. Maybe #13351 is the fix for the underlying issue in crier?

In our own controllers we've *definitely* run into issues and have to be very careful of accidentally performing certain actions twice, and the fact that the cache may not be consistent can influence quite a few design decisions (i.e. giving the things that we are *Creating* deterministic names, so that a second Create will fail with an AlreadyExists error instead of creating 2 instances of the same thing).

I'm missing a bit of context here, but +1 to @alvaroaleman's suggestion that this could well be caused by cache timing issues.

**Pull request closed without merging**